DirectProgramming/C++SYCL_FPGA/Tutorials/Features/max_reinvocation_delay/README.md

```diff
@@ -0,0 +1,214 @@
+
+> **Note**: Even though the Intel® oneAPI DPC++/C++ Compiler is enough to compile for emulation, generating reports and generating RTL, there are extra software requirements for the simulation flow and FPGA compiles.
+>
+> For using the simulator flow, Intel® Quartus® Prime Pro Edition and one of the following simulators must be installed and accessible through your PATH:
+> - Questa*-Intel® FPGA Edition
+> - Questa*-Intel® FPGA Starter Edition
+> - ModelSim® SE
+>
+> When using the hardware compile flow, Intel® Quartus® Prime Pro Edition must be installed and accessible through your PATH.
+
+> **Warning**: Make sure you add the device files associated with the FPGA that you are targeting to your Intel® Quartus® Prime installation.
+
+This sample is part of the FPGA code samples.
+It is categorized as a Tier 2 sample that demonstrates a compiler feature.
+
+```mermaid
+flowchart LR
+   tier1("Tier 1: Get Started")
+   tier2("Tier 2: Explore the Fundamentals")
+   tier3("Tier 3: Explore the Advanced Techniques")
+   tier4("Tier 4: Explore the Reference Designs")
+
+   tier1 --> tier2 --> tier3 --> tier4
+
+   style tier1 fill:#0071c1,stroke:#0071c1,stroke-width:1px,color:#fff
+   style tier2 fill:#f96,stroke:#333,stroke-width:1px,color:#fff
+   style tier3 fill:#0071c1,stroke:#0071c1,stroke-width:1px,color:#fff
+   style tier4 fill:#0071c1,stroke:#0071c1,stroke-width:1px,color:#fff
+```
+
+Find more information about how to navigate this part of the code samples in the [FPGA top-level README.md](/DirectProgramming/C++SYCL_FPGA/README.md).
+You can also find more information about [troubleshooting build errors](/DirectProgramming/C++SYCL_FPGA/README.md#troubleshooting), [running the sample on the Intel® DevCloud](/DirectProgramming/C++SYCL_FPGA/README.md#build-and-run-the-samples-on-intel-devcloud-optional), [using Visual Studio Code with the code samples](/DirectProgramming/C++SYCL_FPGA/README.md#use-visual-studio-code-vs-code-optional), [links to selected documentation](/DirectProgramming/C++SYCL_FPGA/README.md#documentation), and more.
+
+## Key Implementation Details
+
+The sample illustrates the following important concepts.
+
+- How to use the `max_reinvocation_delay` attribute to restrict the maximum delay between loop invocations.
+
+### Description of the `max_reinvocation_delay` Attribute
+
+Apply the `[[intel::max_reinvocation_delay(N)]]` attribute to a loop to restrict the delay between invocations of that loop. The attribute parameter `N` is required and must be a positive constant expression of integer type (currently, only `N=1` is supported, i.e., that there shall be no delay between invocations). This parameter specifies the maximum number of clock cycles allowed between the last iteration of a loop invocation and the first iteration of the next invocation of the loop. The higher the maximum reinvocation delay allowed, the longer the wait before the next loop invocation can start executing.
```

Comment:
Why is the max reinvocation delay not set to the minimum possible value by the compiler to begin with?
As an external user I would think of this max_reinvocation_delay feature to be a "quick fix" for a compiler inefficiency

So I'm guessing there is a reason why it's not the case by default, and I think we should make it clear here.

Reply:
updated! according to the documentation:

"the compiler can optimize the loop better if a high reinvocation delay is allowed in one of the following ways:

 Enable loop speculation
; Pipeline the loop orchestration hardware that can increase fMAX".